src/disk_interface.cc

```diff
@@ -266,22 +266,19 @@ FileReader::Status RealDiskInterface::ReadFile(const string& path,
 
 int RealDiskInterface::RemoveFile(const string& path) {
 #ifdef _WIN32
-  DWORD attributes = GetFileAttributes(path.c_str());
-  if (attributes == INVALID_FILE_ATTRIBUTES &&
-      GetLastError() == ERROR_FILE_NOT_FOUND) {
-    return 1;
-  }
-  if (attributes & FILE_ATTRIBUTE_READONLY) {
-    // On non-Windows systems remove will happily delete read-only files. On
-    // Windows Ninja should behave the same. See
-    // https://github.com/ninja-build/ninja/issues/1886
-    SetFileAttributes(path.c_str(), attributes & ~FILE_ATTRIBUTE_READONLY);
-  }
-  if (!DeleteFile(path.c_str())) {
-    Error("remove(%s): %s", path.c_str(), GetLastErrorString().c_str());
-    return -1;
+  DWORD attributes = GetFileAttributesA(path.c_str());
+  if (attributes == INVALID_FILE_ATTRIBUTES) {
+    DWORD win_err = GetLastError();
+    if (win_err == ERROR_FILE_NOT_FOUND || win_err == ERROR_PATH_NOT_FOUND) {
+      return 1;
+    }
+  } else if (attributes & FILE_ATTRIBUTE_READONLY) {
+    // On non-Windows systems remove() will happily delete read-only files.
+    // On Windows Ninja should behave the same.
+    // See https://github.com/ninja-build/ninja/issues/1886
+    SetFileAttributesA(path.c_str(), attributes & ~FILE_ATTRIBUTE_READONLY);
```

Comment:
If this fails, are we just deferring the error to the actual `DeleteFile` call? Might want to mention the explicit lack of error checking if that's the case.

Reply:
Yes. Note that this was the case before, nothing really to do with this commit (that's why I dislike the added A, because it creates a bigger diff than necessary).

Reply:
I updated the branch to make the changes as very granular commits.

Reply:
I added an explicit comment about the lack of error checking on `SetFileAttributesA`.